packages/legend-graph/src/graphManager/protocol/pure/v1/engine/V1_Engine.ts

```diff
@@ -362,14 +376,24 @@ export class V1_Engine {
 
   // ------------------------------------------- Compile -------------------------------------------
 
-  async compilePureModelContextData(
+  async compilePureModelContextDataWithWarnings(
     model: V1_PureModelContextData,
-    options?: { onError?: (() => void) | undefined } | undefined,
-  ): Promise<void> {
+    options?:
+      | { onError?: (() => void) | undefined; getCompilationWarnings?: boolean }
+      | undefined,
+  ): Promise<EngineWarning[] | undefined> {
     try {
-      await this.engineServerClient.compile(
+      const compileResponse = await this.engineServerClient.compile(
         this.serializePureModelContextData(model),
       );
+      if (
+        compileResponse.warnings &&
+        options?.getCompilationWarnings === true
+      ) {
+        return compileResponse.warnings as EngineWarning[];
```

Comment:
This is not proper, you need to create a shape for `V1_CompilationResult` with proper serialization and a corresponding `CompilationResult`